salesforce/client.bal

```diff
@@ -245,6 +245,73 @@ public isolated client class Client {
         return check self.salesforceClient->delete(path);
     }
 
+    # Lists reports.
+    #
+    # + return - Reports if successful or else `error`
+    @display {label: "List Reports"}
+    isolated remote function listReports() returns @display {label: "Reports"} Report[]|error {
+        string path = utils:prepareUrl([API_BASE_PATH, ANALYTICS, REPORTS]);
+        return check self.salesforceClient->get(path);
+    }
+
+    # Deletes a report.
+    #
+    # + reportId - Report Id
+    # + return - `()` if the report deletion is successful or else an error
+    @display {label: "Delete Report"}
+    isolated remote function deleteReport(@display {label: "Report Id"} string reportId) returns error? {
+        string path = utils:prepareUrl([API_BASE_PATH, ANALYTICS, REPORTS, reportId]);
+        return check self.salesforceClient->delete(path);
+    }
+
+    # Runs an instance of a report synchronously.
+    #
+    # + reportId - Report Id
+    # + return - Reports if successful or else `error`
+    @display {label: "Run Report Synchronously"}
+    isolated remote function runReportSync(@display {label: "Report Id"} string reportId)
+            returns @display {label: "Report Instances"} ReportInstanceResult|error {
```

Comment:
```suggestion
            returns @display {label: "Report Instance"} ReportInstance|error {
```
Can't we do like this as `Runs an instance of a report synchronously` ?

Reply:
Fixed the display annotation correctly

Reply:
ReportInstanceResult --> ReportInstance I guessed, Let's keep as it is for now

Reply:
This was named purposefully.

`ReportInstance` is used for another operation `runReportAsync`. That record will only have few fields about the instance.
But `ReportInstanceResult` will have the full detailed result. Thats why named like this.